server_admin/topics/vault.adoc

```diff
@@ -0,0 +1,57 @@
+[[_vault-administration]]
+
+== Using Vault to Obtain Secrets
+
+Several fields in the administration support obtaining the value of a secret from an external vault.
+
+To obtain a secret from a vault instead of entering it directly, enter
+the following specially crafted string into the appropriate field:
+`**${vault.**_vault.entry.name_**}**` where you replace the `_vault.entry.name_`
+with the name of the secret as recognized by the vault.
+
+Currently, secret can be obtained from vault in the following fields:
+
+SMTP password::
+  In realm <<_email,SMTP settings>>
+
+LDAP bind credential::
+  In <<_ldap,LDAP settings>> of LDAP-based user federation.
+
+OIDC identity provider secret::
+  In _Client Secret_ inside identity provider <<_identity_broker_oidc,OpenID Connect Config>>
+
+To use a vault, a vault provider must be registered within {project_name}.
+It is possible to either use a built-in provider described below or
+implement your own provider. See the link:{developerguide_link}[{developerguide_name}] for more information.
+
+=== Kubernetes / OpenShift plaintext vault provider
+
+{project_name} supports vault implementation for https://kubernetes.io/docs/concepts/configuration/secret/[Kubernetes secrets]. These secrets
+can be mounted as data volumes, and they appear as a directory with flat file structure, where each secret is represented by a file whose name is secret name, and contents of that file is the secret value.
+
+The files within this directory have to be named as secret name prefixed by realm name and an underscore. All underscores within the secret name or realm name have to be doubled in the file name. For example, for a field within a realm called `sso_realm`, a reference to a secret with name `secret-name` would be written as `${vault.secret-name}`, and the file name looked up would be `sso+++__+++realm+++_+++secret-name`.
```

Comment:
sso+++__+++realm+++_+++secret-name - is sso__realm, but I assume it should be sso_realm

Reply:
Here it really needs to be escaped in the file name so that it is distignuishable where the realm name ends.

Reply:
Hmm... after re-reading the paragraph several times I finally understand it, but find it confusing and error prone

Reply:
A different delimiter between realm name and secret name would be better

Reply:
What about secrets/realm-name/secret-name?? Then it's a directory per-realm. Could also have secrets at the top-level that is shared between realms.

Reply:
Sadly this is not user-friendly in K8s, since every entry within a secret would have to have the `path` mapping set for this to work, and I am not certain that OpenShift even supports that from UI or can only set that from within a yaml